### PR TITLE
feat: refactor dialog in favor of input[type=checkbox]

### PR DESCRIPTION
**Pull request by tlgimenes**
## What is this contribution about?
This PR removes the problematic dialog component in favor of the good&old input[type="checkbox"] for both modals and drawers

This PR introduces two new components, Modal.tsx and Drawer.tsx. These components use daisy's modal and drawer classes respectively.

### CODE REVIEW

File: components/header/Drawers.tsx [hunk 1]
```diff
@@ -0,0 +1,94 @@
...
+            title="Minha sacola"
+            onClose={() => displayCart.value = false}
+          >
+            <Cart />
+          </Aside>
+        }
+      >
+        {children}
+      </Drawer>
+    </Drawer>
```
Comment (igorbrasileiro) [inline components/header/Drawers.tsx:90]:
```suggestion
  {children}
</Drawer>
<Drawer // right drawer
        class="drawer-end"
        open={displayCart.value}
        onClose={() => displayCart.value = false}
        aside={
          <Aside
            title="Minha sacola"
            onClose={() => displayCart.value = false}
          >
            <Cart />
          </Aside>
        }
      >
        {children}
      </Drawer>
```

File: components/ui/Drawer.tsx [hunk 2]
```diff
@@ -0,0 +1,61 @@
...
+  const id = useId();
+  const {
+    children,
+    aside,
+    open,
+    onClose,
+    class: _class = "",
+    loading = "lazy",
+  } = props;
+  const [lazy, setLazy] = useState(loading === "lazy" && !open);
```
Comment (igorbrasileiro) [inline components/ui/Drawer.tsx:24]: Wdyt about using preact signals?

File: components/ui/Modal.tsx [hunk 3]
```diff
@@ -1,116 +1,50 @@
...
-  center: "justify-center items-center",
-};
-
-const containerStyles = {
-  "sidebar-right": "h-full w-full sm:max-w-lg",
-  "sidebar-left": "h-full w-full sm:max-w-lg",
-  center: "",
-};
+  useEffect(() => {
+    const handler = (e: KeyboardEvent) =>
```
Comment (igorbrasileiro) [inline components/ui/Modal.tsx:19]: This code seems to be the same of drawer, wdyt about creating a hook for this

File: components/ui/Modal.tsx [hunk 4]
```diff
@@ -1,116 +1,50 @@
...
-          <div class="overflow-y-auto flex-grow flex flex-col">
-            {loading === "lazy" ? lazy.value && children : children}
-          </div>
+    <>
+      <input
+        id={id}
+        checked={open}
+        type="checkbox"
+        class="modal-toggle"
+        onChange={(e) => e.currentTarget.checked === false && onClose?.()}
```
Comment (igorbrasileiro) [inline components/ui/Modal.tsx:38]: Why need this onChange? Is only the css enough, isn't?

File: islands/Header/Buttons.tsx [hunk 5]
```diff
@@ -0,0 +1,3 @@
+export { default as CartButton } from "$store/components/header/Buttons/Cart.tsx";
```
Comment (igorbrasileiro) [inline islands/Header/Buttons.tsx:1]: Remember the bug of fresh related to `export {default}`

File: islands/Header/Drawers.tsx [hunk 6]
```diff
@@ -0,0 +1 @@
+export { default } from "$store/components/header/Drawers.tsx";
```
Comment (igorbrasileiro) [inline islands/Header/Drawers.tsx:1]: Same here

File: components/ui/Modal.tsx (hunk 4 above)
Comment (tlgimenes) [reply components/ui/Modal.tsx:38]: we need to call onClose when the chebox is clicked via the <label /> tag. This works on both islands and non islands components

File: components/ui/Modal.tsx (hunk 3 above)
Comment (tlgimenes) [reply components/ui/Modal.tsx:19]: don't know. Maybe they evolve differently. Let's leave them separated for now

File: components/header/Drawers.tsx (hunk 1 above)
Comment (tlgimenes) [reply components/header/Drawers.tsx:90]: we need to keep the children tree, otherwise we'll have two headers

Comment (tlgimenes) [reply components/header/Drawers.tsx:90]: Doing this will render navbar twice:
image

**Review by igorbrasileiro — APPROVED** [review]

File: islands/Header/Buttons.tsx (hunk 5 above)
Comment (tlgimenes) [reply islands/Header/Buttons.tsx:1]: I'll do in another PR

**Pull request merged**